tableauserverclient/server/endpoint/datasources_endpoint.py

```diff
@@ -24,6 +25,7 @@ class Datasources(Endpoint):
     def __init__(self, parent_srv):
         super(Datasources, self).__init__(parent_srv)
         self._resource_tagger = _ResourceTagger(parent_srv)
+        self._permissions = _PermissionsEndpoint(parent_srv, lambda: self.baseurl)
```

Comment:
@t8y8, you need a lambda because you can't assign the parent baseurl on instantiation because it currently isn't known.  It is only known once you sign in and the library determines what API to use.

This problem was solved separately in `_ResourceTagger` by taking in a `baseurl` every single time you call any member function.  I didn't like this model because since the object is owned by a parent endpoint, its `baseurl` should be part of the object.  I guess an alternative here could be requiring the parent to provide itself to this object's constructor, but I felt that produced unnecessary coupling.

Reply:
for reference, below is a snippet from `_ResourceTagger`

```
    # Remove and add tags to match the resource item's tag set
    def update_tags(self, baseurl, resource_item):
        if resource_item.tags != resource_item._initial_tags:
            add_set = resource_item.tags - resource_item._initial_tags
            remove_set = resource_item._initial_tags - resource_item.tags
            for tag in remove_set:
                self._delete_tag(baseurl, resource_item.id, tag)
            if add_set:
                resource_item.tags = self._add_tags(baseurl, resource_item.id, add_set)
            resource_item._initial_tags = copy.copy(resource_item.tags)
        logger.info('Updated tags to {0}'.format(resource_item.tags))
```

As you can see, `baseurl` is a parameter to the member function

Reply:
Got it, thank you! We may want to refactor _ResourceTagger to do the same #notetoself